src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsPathContext2.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// NuGet path information specific to the current context (e.g. project context) or solution context
+    /// Represents captured snapshot associated with current project/solution settings.
+    /// Should be discarded immediately after all queries are done.
+    /// </summary>
+    [ComImport]
+    [Guid("24A1A187-75EE-4296-A8B3-59F0E0707119")]
+    public interface IVsPathContext2 : IVsPathContext
```

Comment:
Should this interface have a more descriptive name?

Reply:
that's the pattern being followed, not just NuGet but across VS. It's easy to relate to earlier interface and the fact that new interface is a build up on top of existing one.